evm/src/executor/builder.rs

```diff
@@ -53,8 +53,10 @@ impl Fork {
     pub async fn spawn_backend(self, env: &Env) -> SharedBackend {
         let Fork { cache_path, url, pin_block, chain_id } = self;
 
-        let provider =
-            Arc::new(Provider::try_from(url.clone()).expect("Failed to establish provider"));
+        let provider = Arc::new(
+            Provider::<RetryClient<Http>>::new_client(url.clone().as_str(), 10, 1)
```

Comment:
needs to reflect milliseconds change, since we have exponential, we can probably start with 1000ms?